### ISSUE TITLE
Pages sometimes take a minute or more to load

### ISSUE THREAD

**Issue by egermani** [neo4j-examples/neo4j-movies-template#41]
Hi there,

First off, thanks for this app. I wanted to learn React and got more than I bargained for here -- cool to see how Redux works, too. I've been customizing the app, and started to notice that some percentage of requests (maybe 15-20%) take an incredibly long time to resolve. The response payload is 369B, so that shouldn't be an issue. I'm not sure how to debug this. Things I've tried:

- Stripping out all the code from the render function that required information from the database. Still slow.
- Using a profiler from Werkzeug, which apparently Flask runs on. This didn't work because the middleware that wrapped the app wasn't compatible with all the Swagger stuff.
- Disabling IPv6 on my localhost -- apparently this can be a problem with Werkzeug. No dice there, either.

Anyone have suggestions on how to figure this out? Thanks in advance!

**Comment by jexp** [comment]
@whatSocks any ideas?

**Comment by egermani** [comment]
Hey! I'm going to shoot another flare on this one, because I'm still very excited about my idea for this app and this issue is proving to be a backbreaker on my progress.

My current best guess is that there's some limitation on how many requests Chrome will handle or acknowledge in a short timeframe, because if I try clicking around quickly, the third or fourth XHR request will hang for the 45+ seconds. I'm getting into pure conjecture here, because I don't know much of anything about networking, but perhaps these XHR requests (tiny though they are) are not being concluded and garbage-collected quickly enough, resulting in some kind of traffic jam.

Does this seem likely? If so, what could I do?

Thanks!

**Comment by whatSocks** [comment]
Hey @egermani we're looking into it. I guess your previous messages got buried in my inbox

**Comment by egermani** [comment]
No worries, I appreciate you taking some time for this issue -- I understand it may be tricky to diagnose. I'm going to try serving the Flask app through Gunicorn and see if that helps any, I'll report back here if it does.

**Comment by egermani** [comment]
So, Gunicorn proved not to be the savior, though I got a little more info out of the debug logs. When the page was hanging, Gunicorn reported that its worker had timed out and needed to respawn. Why it timed out? Unclear.

I profiled one of the app methods that has a call to the database in it, and it never took more than half a second to complete -- but that was done by executing the `app.py` script with a profiler snippet at the bottom. I did this because I read that worker timeouts are tied to long running processes, and I wanted to double-check that my methods weren't somehow taking forever.

I tried looking at the Gunicorn logs, but when I routed that to my systemlog nothing from Gunicorn showed up -- probably wasn't calling that setting correctly.
